### PR TITLE
WIP: replace some resources get via get_url with local copy

### PR DESCRIPTION
**Pull request by ingvagabund**
Minimize reliance on remote resources, use local ones instead.

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/contrib/1621)

### CODE REVIEW

**Comment by ingvagabund** [comment]
@redhat-k8s-bot test please

**Comment by ingvagabund** [comment]
Tested within the same CI with "kube_source_type=github-release kube_version=1.2.4". Kubernetes got deployed.

**Comment by eparis** [comment]
I hate having two places for these things to live and to get out of date...
/lgtm

**Comment by ingvagabund** [comment]
@eparis I don't like it either. What about to completely remove init/systemd and move its content under corresponding files directories of master/node/kubernetes under ansible? There is no difference if people get the service and config files from ``init/systemd`` or from ``ansible/roles/[master|node|kubernetes]/files``.

File: ansible/roles/master/tasks/retrieve-systemd-files.yml [hunk 1]
```diff
@@ -1,9 +1,6 @@
 ---
-- name: Get Systemd unit files from Kubernetes repository
-  get_url:
-    url=https://raw.githubusercontent.com/kubernetes/contrib/master/init/systemd/{{ item }}.service
```
Comment (sttts) [inline ansible/roles/master/tasks/retrieve-systemd-files.yml]: Are those files now duplicated in the contrib repo?

Comment (ingvagabund) [inline ansible/roles/master/tasks/retrieve-systemd-files.yml]: They are. However, the contrib repository is planned to be retired. So we need to find new place for all the systemd and config files.

**Comment by ingvagabund** [comment]
The config files are meant for ``1``-node cluster anyway [1]. Though, it does not make much sense to use the ansible to deploy the ``1``-node cluster, it generates tokens and launches kubernetes-addons. Which is the missing piece when you install kubernetes from rpms. Still not a plug&play solution [2].

In short:
**environ & config files**
* `config` and `kubernetes.conf` are under kubernetes role as it is common for all components
* `apiserver`, `controller-manager` and `scheduler` belong to master components only (plus with `kubelet` in some cases)
* `kubelet` and `proxy` belong to node components only

**service files**
* `kube-apiserver.service`, `kube-controller-manager.service`, `kube-scheduler.service` belongs to master components (plus with `kubelet.service` in some cases)
* `kubelet.service` and `kube-proxy.service` belongs to node components

Service files are not templated so far. Only configs are. And afaik, all the non-templated config files are used in distribution builds only.

But still, we will have at duplicates. Until resolved, let's not merge the PR.

[1] https://github.com/kubernetes/contrib/tree/master/init/systemd
[2] https://github.com/kubernetes/contrib/issues/1325

**Comment by ingvagabund** [comment]
Or another option is to

1. move init/systemd under ansible directory
1. in every occurrence of the systemd service of config file check if "../init/systemd/[NAME.service|environ/NAME] file exists. If it does, copy it, if it does not, download it.

It will not change behaviour of the current PR. At the same time when each role gets publish in Ansible galaxy, it will work. And, there will be no (obvious) duplicates. But still the files get downloaded unless the roles are used directly from the contrib repository.

There is already some duplication between templated and non-templated config files. However, it should not matter as the non-templated configs are used purely for `1`-node cluster and can be independent of templated ones (unless some fundamental thing needs to change).

**Comment by sttts** [comment]
If the other files outside of ansible have a different purpose and life-cycle, I don't care about duplicates. But a self contained ansible directory without unversioned files downloaded with http(s) is a good goal.

**Comment by test-foxish** [comment]
recomputing cla status...

**Comment by grodrigues3** [comment]
[APPROVALNOTIFIER] The Following OWNERS Files Need Approval:

- **ansible**

We suggest the following people:
cc @bprashanth 
 You can indicate your approval by writing `/approve` in a comment
 You can cancel your approval by writing `/approve cancel`in a comment

**Pull request closed without merging**